### ISSUE TITLE
Templates from extensions are not registered in the Flask app

### ISSUE THREAD

**Issue by amercader** [ckan/ckan#3833]
This only affects maser. The best way to reproduce it is to click on the Data API button on a resource with DataStore. This uses the API to look for a template and return it rendered, and right now triggers a 500 error, as the template comes from the datastore extension:

```
  File "/home/adria/dev/pyenvs/ckan/src/ckan/ckan/views/api.py", line 448, in snippet
    return render(snippet_path, extra_vars=dict(request.args))
  File "/home/adria/dev/pyenvs/ckan/src/ckan/ckan/lib/base.py", line 187, in render
    return flask_render_template(template_name, **extra_vars)
  File "/home/adria/dev/pyenvs/ckan/local/lib/python2.7/site-packages/flask/templating.py", line 133, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "/home/adria/dev/pyenvs/ckan/local/lib/python2.7/site-packages/jinja2/environment.py", line 869, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "/home/adria/dev/pyenvs/ckan/local/lib/python2.7/site-packages/jinja2/environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/home/adria/dev/pyenvs/ckan/local/lib/python2.7/site-packages/jinja2/environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "/home/adria/dev/pyenvs/ckan/local/lib/python2.7/site-packages/jinja2/loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "/home/adria/dev/pyenvs/ckan/local/lib/python2.7/site-packages/flask/templating.py", line 57, in get_source
    return self._get_source_fast(environment, template)
  File "/home/adria/dev/pyenvs/ckan/local/lib/python2.7/site-packages/flask/templating.py", line 85, in _get_source_fast
    raise TemplateNotFound(template)
TemplateNotFound: ajax_snippets/api_info.html
```

Perhaps we need to add something to the [`add_template_directory()`](https://github.com/ckan/ckan/blob/master/ckan/plugins/toolkit.py#L326) function that plugins use

cc @tino097 @smotornyuk

**Comment by tino097** [comment]
@amercader im not able to reproduce the error

**Comment by wardi** [comment]
This looks like the latest error I'm getting in circleci https://circleci.com/gh/ckan/ckan/3217#tests/containers/2

**Comment by amercader** [comment]
Fixed in #3860
